src/main/java/com/google/sps/data/User.java

```diff
@@ -44,8 +44,12 @@ else if(input == "mentor" || input == "MENTOR") {
   }
 
 
-  public boolean equals(User user) {
-    return this.name == user.name;
+  public boolean equals(Object user) {
+    if (user instanceof User) {
+        User toCompare = (User) user;
+        return this.name.equals(toCompare.name);
```

Comment:
would it make sense to use the User.id for this comparison?

Reply:
We are planning on doing that in the future, but because all the IDs are zero right now, it wouldn't work for this pull request.